traits/tests/test_trait_set_object.py

```diff
@@ -190,6 +333,53 @@ def test_ixor(self):
         self.assertEqual(self.added, {5})
         self.assertSetEqual(ts, {5})
 
+    def test_ixor_no_nofications_for_no_change(self):
+        notifier = mock.Mock()
+        ts_1 = TraitSet([1, 2], notifiers=[notifier])
+
+        # when
+        ts_1 ^= set()
+
+        # then
+        notifier.assert_not_called()
+
+    def test_ixor_with_iterable_items(self):
+        iterable = range(2)
+
+        python_set = set([iterable])
+        python_set ^= set([iterable])
+        self.assertEqual(python_set, set())
+
+        ts = TraitSet([iterable], validator=self.validator)
+        self.validator_args = None
+        ts ^= [iterable]
+        self.assertEqual(ts, set())
+
+        # No values are being added.
+        self.assertIsNone(self.validator_args)
+
+    def test_ixor_validator_args_with_added(self):
+        # Check the values given to the validator
+        # when symmetric_difference_update is called.
+
+        validator_args = None
+
+        def validator(set_, added):
+            nonlocal validator_args
+            validator_args = (set_, added)
+
+            return set(str(value) for value in added)
+
+        ts = TraitSet([1, 2, 3], validator=validator)
+        self.assertEqual(ts, set(["1", "2", "3"]))
+
+        # when
+        ts ^= set(["2", 3, 4])
+
+        # then
+        self.assertEqual(validator_args, (ts, set([3, 4])))
+        self.assertEqual(ts, set(["1", "4"]))
```

Comment:
Ha, I neglected to check the values provided to the notifier for this scenario, and I bet the values are misleading (`added` is set(["3", "4"]) instead of just `set(["4"])`. I will have to come back to fix this next week.